core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -78,7 +83,9 @@ public void start() throws Exception {
       mLeaderSelector.waitForState(State.PRIMARY);
       if (gainPrimacy()) {
         mLeaderSelector.waitForState(State.SECONDARY);
-        losePrimacy();
+        if (mRunning) {
```

Comment:
when mRunning is false, how do we exit the while loop?

Reply:
Add the `if(mRunning) {losePrimacy()} else {break the while loop}", thanks!